scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -68,14 +73,51 @@ object AsyncFlowDefRunner {
 }
 
 /**
+ * This holds an internal thread to run
  * This holds an internal thread to submit run
  * a Config, Mode, FlowDef and return a Future holding the
  * JobStats
  */
 class AsyncFlowDefRunner extends Writer { self =>
   import AsyncFlowDefRunner._
 
-  private[this] val filesToCleanup = mutable.Set[String]()
+  private[this] val mutex = new AnyRef
```

Comment:
personal OCD. I don't like to lock things others can see. I always like to lock a private val.